### DISCUSSION TITLE
Filecoin application ideas

### DISCUSSION THREAD

**Discussion by rjan90** [filecoin-project/community#168] [General]
Hey! :wave:

Thought it would be cool to have a thread where we could brainstorm possible applications/tools made on top of Filecoin! I think it might be helpful for people looking to develop an app on Filecoin to see what is potentially possible, apart from just storing files. So if you have any ideas you have been thinking about, please post them here. Also add if there is anything specific that needs to be developed (like a smart-contract on Filecoin) before it is possible to have such an app :smile: 

From the [Filecoin <-> Chainlink integration blog post](https://filecoin.io/blog/posts/filecoin-and-chainlink-integration/), a couple of possible applications:

1. Filecoin Miner Insurance - Chainlink oracles can monitor the current state of Filecoin miners, and if they go offline or cannot successfully retrieve data from storage, an alert is relayed to an insurance smart contract on another blockchain to automatically execute payouts to policyholders. For example, storing NFT data on Filecoin, with on-chain punishment if it cannot be retrieved in a timely manner.

2. DataDAOs — A DataDAO, or a DAO holding extensive datasets like historical price databases or research data, can leverage Filecoin to store their datasets. Chainlink oracles can then deliver portions of the data to smart contracts on other blockchains, creating new revenue streams for DataDAOs.

3. Data Bounties — Smart contracts that request certain datasets as characterized by their unique CIDs can be stored on Filecoin. Payments can be triggered when the dataset is verifiably proven on Filecoin and relayed through Chainlink.

4. DeFi Data — Leveraging Filecoin for cheaper data storage, Chainlink can enable a large amount of financial market data to be stored on Filecoin long-term, which is then delivered to other blockchains on-demand to support more advanced DeFi applications.

5. Automated Storage — Smart contracts can use Chainlink oracle services to automatically store some type of data on Filecoin at specific intervals, creating tamper-proof and highly efficient data archives. For example, information such as payout schedules and bonding curves—which provide key insights for investors but take up a large amount of storage—can be automatically stored on Filecoin every so many blocks.

**Comment by funktafide** [comment]
I hate doing KYC at every new place I have to sign up for. It would be great if there was a universally accepted KYC validation service where you could do KYC once with them and store your KYC information on Filecoin. When you sign up for a new site you use the accepted KYC service and your verification is done. If the data needs to be manually verified they can pull your data from Filecoin. This way you don't have to store this data on multiple websites. Just a thought I was dreaming up 😀. I don't have the developer experience to know how feasible it would be.

**Comment by flyworker** [comment]
It is interesting, but do we have a timeline for Filecoin smart contracts?
or we need other blockchain as a bridge?

**Reply by rjan90** [reply]
There is apparently a prototype of an [ERC-20 style actor on Filecoin](https://filecoinproject.slack.com/archives/C0179RNEMU4/p1613089672422200?thread_ts=1613024526.379600&cid=C0179RNEMU4). But I don't think it is  going to be implemented until the FIP's for reducing on-chain congestion are deployed to the network.
